src/components/LabelPicker/LabelPicker.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+
+export type Label = {
+    /** the display name to be shown to the user */
+    name: string;
+    /** state of the label: selected or not */
+    isSelected: boolean;
+    /** the number of occurrences will be shown next to the name */
+    count?: number;
+};
+
+export interface Props<B = HTMLButtonElement> {
+    /** a list of available labels and their attributes */
+    labels: Label[];
+    /** a button like element that supports onClick handler to be used as the trigger */
+    children: React.ReactElement<React.ButtonHTMLAttributes<B>>;
+    /** callback to be called when the state of a checkbox changes */
+    onChange: (Label) => void;
+    /** callback to add new label */
+    onAdd: (name: string) => void;
+    /** callback when clicking the Done button */
+    onDone?: () => void;
+    /** callback on outer click */
+    onCancel?: () => void;
+    /** text to be shown as input placeholder for adding labels */
+    inputPlaceholder: string;
+    /** label for the done button */
+    doneLabel: string;
+    /** Ref to access the input field */
+    ref?: React.ForwardedRef<HTMLButtonElement>;
+}
```

Comment:
I'd also add some validation message prop, e.g `validationMessage`. For instance, if a typed value already exists in labels or a value are too short. In this scenario, some warning message will be shown, and add button will be disabled. What do you think?

Reply:
do you mean you want to validate the name of the new label somehow? I think that can be done on the application side all together, since the actual adding happens on the user side (see story)

Reply:
I agree that validation is a part of business logic, but I'm thinking about how we can notify the users that something is wrong with typed values. Probably we need to show some warning message in the label's popup. Also, in this scenario, we need to disable `Add` button.

Reply:
after f2f discussion decided to leave it as it is